libs/ui-toolkit/src/components/input/input.tsx

```diff
@@ -0,0 +1,93 @@
+import { InputHTMLAttributes, forwardRef } from 'react';
+import classNames from 'classnames';
+import { Icon, IconName } from '../icon';
+
+interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  hasError?: boolean;
+  disabled?: boolean;
+  className?: string;
+  prependIconName?: IconName;
+  appendIconName?: IconName;
+}
+export const inputClassNames = ({
+  hasError,
+  className,
+}: {
+  hasError?: boolean;
+  className?: string;
+}) =>
+  classNames(
+    [
+      'inline-flex',
+      'items-center',
+      'box-border',
+      'border',
+      'bg-clip-padding',
+      'border-black-60 dark:border-white-60',
+      'bg-black-25 dark:bg-white-25',
+      'text-black-60 dark:text-white-60',
+      'text-ui',
+      'focus-visible:shadow-focus dark:focus-visible:shadow-focus-dark',
+      'focus-visible:outline-0',
+      'disabled:bg-black-10 disabled:dark:bg-white-10',
+    ],
+    {
+      'pl-8': !className?.match(/(^| )p(l|x)-\d+( |$)/),
```

Comment:
@bglownia  is this regex simply to override the padding to pl-8 if someone supplies padding. If so I'd almost prefer to not allow className as a prop to the component.

I think it might help restrict the design system so we more carefully consider when to add further variations.

Reply:
I've put the regexes into variables which I do think make things easier to read